altair/utils/deprecation.py

```diff
@@ -1,94 +1,113 @@
 from __future__ import annotations
 
 import sys
-from typing import Callable, TypeVar, TYPE_CHECKING
 import warnings
-import functools
+from typing import TYPE_CHECKING
 
-if sys.version_info >= (3, 10):
-    from typing import ParamSpec
+if sys.version_info >= (3, 13):
+    from warnings import deprecated as _deprecated
 else:
-    from typing_extensions import ParamSpec
+    from typing_extensions import deprecated as _deprecated
+
 
 if TYPE_CHECKING:
-    from functools import _Wrapped
+    if sys.version_info >= (3, 11):
+        from typing import LiteralString
+    else:
+        from typing_extensions import LiteralString
 
-T = TypeVar("T")
-P = ParamSpec("P")
-R = TypeVar("R")
 
+class AltairDeprecationWarning(DeprecationWarning): ...
 
-class AltairDeprecationWarning(UserWarning):
-    pass
 
+def _format_message(
+    version: LiteralString,
+    alternative: LiteralString | None,
+    message: LiteralString | None,
+    /,
+) -> LiteralString:
+    output = f"Deprecated in `altair={version}`."
+    if alternative:
+        output = f"{output} Use {alternative} instead."
+    return f"{output}\n{message}" if message else output
 
+
+# NOTE: Annotating the return type breaks `pyright` detecting [reportDeprecated]
```

Comment:
I hope my comments here make sense. 

There is some strange behaviour but this specific combination of quirks makes `pyright` happy